hera_cal/abscal.py

```diff
@@ -3436,29 +3443,62 @@ def post_redcal_abscal(model, data, data_wgts, rc_flags, edge_cut=0, tol=1.0, ke
                                              data_wgts=data_wgts, tol=tol, keep_flagged_ants=True)
     reds = redcal.get_reds(idealized_antpos, pols=data.pols(), bl_error_tol=redcal.IDEALIZED_BL_TOL)
 
+    # separate autos and crosses if running amplitude calibration with autos only:
+    if autos_only_abs_amp_logcal:
+        autos = [antpol for antpol in data.keys() if antpol[0] == antpol[1]]
+        autos_data = DataContainer({antpol:data[antpol] for antpol in autos})
+        autos_data_wgts = DataContainer({antpol:data_wgts[antpol] for antpol in autos})
+        autos_model = DataContainer({antpol:model[antpol] for antpol in autos})
+
+        #separate crosses from autos:
+        crosses = [antpol for antpol in data.keys() if antpol[0] != antpol[1]]
+        crosses_data = DataContainer({antpol:data[antpol] for antpol in crosses})
+        crosses_data_wgts = DataContainer({antpol:data_wgts[antpol] for antpol in crosses})
+        crosses_model = DataContainer({antpol:model[antpol] for antpol in crosses})
+
     # Abscal Step 1: Per-Channel Logarithmic Absolute Amplitude Calibration
-    gains_here = abs_amp_logcal(model, data, wgts=data_wgts, verbose=verbose, return_gains=True, gain_ants=ants)
+    if autos_only_abs_amp_logcal:
+        gains_here = abs_amp_logcal(autos_model, autos_data, wgts=autos_data_wgts, verbose=verbose, return_gains=True, gain_ants=ants)
+    else:
+        gains_here = abs_amp_logcal(model, data, wgts=data_wgts, verbose=verbose, return_gains=True, gain_ants=ants)
+     
     abscal_delta_gains = {ant: gains_here[ant] for ant in ants}
     apply_cal.calibrate_in_place(data, gains_here)
 
+
     # Abscal Step 2: Global Delay Slope Calibration
-    binary_wgts = DataContainer({bl: (data_wgts[bl] > 0).astype(float) for bl in data_wgts})
+    if autos_only_abs_amp_logcal:
+        binary_wgts = DataContainer({bl: (data_wgts[bl] > 0).astype(float) for bl in crosses_data_wgts})
+    else:
+        binary_wgts = DataContainer({bl: (data_wgts[bl] > 0).astype(float) for bl in data_wgts})
     df = np.median(np.diff(data.freqs))
     for time_avg in [True, False]:  # first use the time-averaged solution to try to avoid false minima
-        gains_here = delay_slope_lincal(model, data, idealized_antpos, wgts=binary_wgts, df=df, f0=data.freqs[0], medfilt=True, kernel=kernel,
+        if autos_only_abs_amp_logcal:
```

Comment:
Having a separate call to each calibration function depending on whether `autos_only_abs_amp_logcal` is true or makes the code a lot more complex and annoying to test and maintain.